DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1CompareVertexSoA.cc

```diff
@@ -0,0 +1,173 @@
+    auto zc = vsoaCPU.zv[sic];
+    auto xc = x0 + dxdz * zc;
+    auto yc = y0 + dydz * zc;
+    zc += z0;
+
+    auto ndofCPU = vsoaCPU.ndof[sic];
+    auto chi2CPU = vsoaCPU.chi2[sic];
+
+    const int32_t notFound = -1;
+    int32_t closestVtxidx = notFound;
+    float mindz = dzCut_;
+
+    for (int ivg = 0; ivg < nVerticesGPU; ivg++) {
+      auto sig = vsoaGPU.sortInd[ivg];
+      auto zgc = vsoaGPU.zv[sig] + z0;
+      auto zDist = std::abs(zc - zgc);
+      //insert some matching condition
+      if (zDist > dzCut_)
+        continue;
+      if (mindz > zDist) {
+        mindz = zDist;
+        closestVtxidx = sig;
+      }
+    }
+    if (closestVtxidx == notFound)
+      continue;
+
+    auto zg = vsoaGPU.zv[closestVtxidx];
+    auto xg = x0 + dxdz * zg;
+    auto yg = y0 + dydz * zg;
+    zg += z0;
+    auto ndofGPU = vsoaGPU.ndof[closestVtxidx];
+    auto chi2GPU = vsoaGPU.chi2[closestVtxidx];
+
+    hx_->Fill(xc, xg);
+    hy_->Fill(yc, yg);
+    hz_->Fill(zc, zg);
+    hchi2_->Fill(chi2CPU, chi2GPU);
+    hchi2oNdof_->Fill(chi2CPU / ndofCPU, chi2GPU / ndofGPU);
+    hptv2_->Fill(vsoaCPU.ptv2[sic], vsoaGPU.ptv2[closestVtxidx]);
+    hntrks_->Fill(ndofCPU + 1, ndofGPU + 1);
+  }
+  hnVertex_->Fill(nVerticesCPU, nVerticesGPU);
+}
+
+//
+// -- Book Histograms
+//
+void SiPixelPhase1CompareVertexSoA::bookHistograms(DQMStore::IBooker& ibooker,
+                                                   edm::Run const& iRun,
+                                                   edm::EventSetup const& iSetup) {
+  //std::string top_folder = ""//
+  ibooker.cd();
+  ibooker.setCurrentFolder(topFolderName_);
+  hnVertex_ = ibooker.book2D("nVertex", "# of Vertex;CPU;GPU", 101, -0.5, 100.5, 101, -0.5, 100.5);
+  hx_ = ibooker.book2D("vx", "Vertez x;CPU;GPU", 20, -0.1, 0.1, 20, -0.1, 0.1);
+  hy_ = ibooker.book2D("vy", "Vertez y;CPU;GPU", 20, -0.1, 0.1, 20, -0.1, 0.1);
+  hz_ = ibooker.book2D("vz", "Vertez z;CPU;GPU", 30, -30., 30., 30, -30., 30.);
+  hchi2_ = ibooker.book2D("chi2", "Vertex chi-squared;CPU;GPU", 40, 0., 20., 40, 0., 20.);
+  hchi2oNdof_ = ibooker.book2D("chi2oNdof", "Vertex chi-squared/Ndof;CPU;GPU", 40, 0., 20., 40, 0., 20.);
+  hptv2_ = ibooker.book2D("ptsq", "Vertex p_T squared;CPU;GPU", 200, 0., 200., 200, 0., 200.);
+  hntrks_ = ibooker.book2D("ntrk", "#tracks associated;CPU;GPU", 100, -0.5, 99.5, 100, -0.5, 99.5);
+  hntrks_ = ibooker.book2D("ntrk", "#tracks associated;CPU;GPU", 100, -0.5, 99.5, 100, -0.5, 99.5);
+}
+
+void SiPixelPhase1CompareVertexSoA::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // monitorpixelVertexSoA
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("pixelVertexSrcCPU", edm::InputTag("pixelVerticesSoA@cpu"));
+  desc.add<edm::InputTag>("pixelVertexSrcGPU", edm::InputTag("pixelVerticesSoA@cuda"));
+  desc.add<edm::InputTag>("beamSpotSrc", edm::InputTag("offlineBeamSpot"));
+  desc.add<std::string>("topFolderName", "SiPixelHeterogeneous/PixelVertexCompareSoAGPU vs CPU");
```

Comment:
```suggestion
  desc.add<std::string>("topFolderName", "SiPixelHeterogeneous/PixelVertexCompareSoAGPUvsCPU");
```
Having a space in the folder name _might_ lead to problems somewhere downstream ?
I don't really know, but I'm not used to seeing spaces in the folder names.